### ISSUE TITLE
Tutorial for beginner

### ISSUE THREAD

**Issue by ymys** [Ardakilic/Telegram-bot-php#5]
Sorry for asking this, but you said: By installing this app, you can have a working Telegram Bot within minutes!

I'd already follow your instructions, but stuck after eidting config/main.php. How could I navigate to http://yoursite.com/set_webhook/**bothook**? It said error 404.

**Comment by Ardakilic** [comment]
Hello,

I'll try to describe the steps one by one:
- First, clone the project into your server and run `composer install` in its directory. Then the dependencies will be installed. 

If you see composer command not found in the terminal, head over to [Composer getting started guide](https://getcomposer.org/doc/00-intro.md) and install it, and run the command again.
- Secondly, you need a running web server that the root is set to `public` folder, and which should have routes redirected to index.php (think like Laravel's or Wordpress's pretty URLs feature), since this app is build with Silex, you can refer to [Silex's official documentation](http://silex.sensiolabs.org/doc/web_servers.html) for this. There are examples for Apache and Nginx web servers in the link.

The web server should also serve from https (restriction of Telegram) protocol. if you're using Cloudlare DNS, you can have free [Flexible SSL](https://www.cloudflare.com/ssl) which should also work here. Else you need to buy an SSL certificate and install it into your web server. (Self signed SSL certificates don't work!)

To make sure your web server configuration is running, you can make a simple test:

Add this into your `public/index.php` before `$app->run()` last line:

``` php
$app->get('/test', function() use($app) {
    return "Test is running, you are awesome!";
});
```

And navigate to http**s**://yoursite.com/test.

If you see "_Test is running, you are awesome!_" in the browser, you set the web server right.
- Thirdly, you need to edit the `config/main.php` file.

Let's assume you have a bot called @**ymysBot**. First, you need an alias (which will be served as route parameter, because this app can handle multiple Telegram bots simultaneously and the alias is what defines to which bot Telegram sends response to) for this.  

Let's say alias is **ymys** for @**ymysBot**, then `config/main.php`'s `bots` key should be like:

``` php
//in config/main.php
'bots' => [
    'ymys' => 'ymysBot',
],
```
- Now, you need to add your bot's own configuration file. 

Copy or rename `config/bots/myDemoBot.php` as `config/bots/ymysBot.php` (the name of the bot as file name, not the alias), and fill your API key, and bot responses. You can get the api key from @**Botfather** from Telegram messenger directly.
- Now you need to set up the webhook and teach Telegram where it should send your bot requests to:

Navigate through http(s)://yoursite.com/set_webhook/**ymys** (ymys = the alias which we've decided moments ago). If everything went okay, you will see a response from Telegram which tells webhook was set.
- Now you can test your bot. Assuming you left your responses as is from `config/myDemoBot.php` you have two responses: _message1_ and _hodor_ (please refer to [this link](https://github.com/Ardakilic/Telegram-bot-php/blob/0515280efc6ef79755dd62f42b9329ec1ea41840/config/bots/myDemoBot.php#L26-L40) and check the highlighted lines if you've lost me.)

Now either try sending @**ymysBot** a message directly or add it to a conversation group.

If you are messaging with the bot directly, you can directly send _message1_ or _hodor_ as text.

If the bot is in a conversation group, you need to mention it and send the message, like this:

@**ymysBot message1** or @**ymysBot hodor**, the will send the response to the chat.

(p.s: If you typed these, if the hook is set, in the backend, Telegram API sends a POST request to http**s**://yoursite.com/hook/**ymys** , this is the reason for the alias and SSL)

### tl;dr version:
- Download and install dependencies,
- Set up your web server.
- Set an alias for your bot in config/main.php.
- Make a config/bots/youtbot.php file from provided template and fill messages and api key for the bot.
- Navigate to http_s_://yourserver.com/set_webhook/_yourbotalias_
- Start chatting with the bot either in a group or via direct messages.

I hope this message clarifies it a bit. I will try to create a Wiki page after I've completed SQL integration. Nowadays I'm busy with my own work so I cannot find time for them.

**Comment by Ardakilic** [comment]
Closing this due to inactivity and release of the new version. Please open a new issue when you encounter an error with the new version.
